cli/src/api/mod.rs

```diff
@@ -573,6 +589,66 @@ impl PhylumApi {
     }
 }
 
+/// Phylum group types.
+pub enum Group {
+    Legacy(String),
+    Org(OrgGroup),
+}
+
+impl Group {
+    /// Create a new group.
+    fn new(org: String, group: String) -> Self {
+        Self::Org(OrgGroup { org, name: group })
+    }
+
+    /// Create a group without organization.
+    fn new_legacy(group: String) -> Self {
+        Self::Legacy(group)
+    }
+
+    /// Create a group from an optional group name.
+    pub fn try_new<O, G>(org: Option<O>, group: Option<G>) -> Option<Self>
+    where
+        O: Into<String>,
+        G: Into<String>,
+    {
+        group.map(|group| match org {
+            Some(org) => Self::new(org.into(), group.into()),
+            None => Self::new_legacy(group.into()),
+        })
+    }
```

Comment:
Should these be "from" functions instead of "new" functions? They're just moving the input parameters into enum variants. Maybe we don't need `new` or `new_legacy` at all. I also wonder, if we're not having a `None` variant, would it be better to just have `struct Group { organization: Option<String>, name: String }` instead of an enum.

Reply:
Yes, these should be `new` functions. 'From' impls on random tuples or strings don't make sense. I don't think changing anything here would improve things.

Reply:
I'd rather `Group::from_legacy` or `Group::legacy` or just `Group::Legacy` over `Group::new_legacy`.

Reply:
`Group::Legacy` or `Group::legacy` are both not really an option, since they stray from Rust standards.

`Group::from_legacy` implies what you pass is a legacy group, which could be argued but really at that point you'd expect it to be `Group::Legacy` as parameter.

Because of that `Group::new_legacy` is the only option that is both accurate and follows common rust patterns.

Reply:
`Group::Legacy(name)` is the Rust standard for enums, and we already have it. I don't see why we would also need `Group::new_legacy(name)` which does exactly the same thing, except maybe to avoid writing `Group::Org(OrgGroup { organization, name })`, which I still believe to be needless complication.

However, neither `Group::new_legacy` nor `Group::new` are ever called. We probably don't need them.